src/openforms/submissions/form_logic.py

```diff
@@ -124,7 +129,35 @@ def evaluate_form_logic(
                     if submission_step_to_modify == step:
                         updated_data = {}
                         step._is_applicable = False
+                elif action_details["type"] == LogicActionTypes.variable:
+                    new_value = jsonLogic(action_details["value"], data)
+                    submission_variable = submission_variable_state.get_variable(
+                        action["variable"]
+                    )
+                    updated_data[action["variable"]] = new_value
+
+                    form_variable = submission_variable.form_variable
+
+                    if (
+                        not form_variable
+                        or not form_variable.form_definition
+                        or form_variable.form_definition
+                        != step.form_step.form_definition
+                    ):
+                        # Cases where either:
+                        # 1. The variable is not related to a particular step
+                        # 2. The variable being changed is part of a different step than the step currently being edited
+                        # In these cases, the changes are persisted to the database
+                        submission_variable.value = new_value
+                        submission_variable.source = (
+                            SubmissionValueVariableSources.logic
+                        )
+                        submission_variables_to_update.append(submission_variable)
```

Comment:
we should not be able to do that when doing a logic check with dirty data though, right?